### PR TITLE
Raise appropriate exception when upload fails.

### LINKED ISSUE #1712: File upload to Storage fails silently
Hello, I'm trying to upload a file to Storage from a Compute instance.
Following the example in the [docs](https://googlecloudplatform.github.io/gcloud-python/stable/#cloud-storage):
```
from gcloud import storage
client = storage.Client()
bucket = client.get_bucket('my-bucket-name')
blob = bucket.blob('my-test-file.txt')
blob.upload_from_string('this is test content!')
```
Nothing happens. No error message, no file created/uploaded in the bucket, `blob.exists()` returns `False`.

I don't think it's an access control/permissions thing as the bucket has default settings and I'm able to modify existing blobs.

Any leads on how to debug this?

Also, I think it would make sense to throw an exception or give some sort of indication/explanation that it failed.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1712.

### CODE REVIEW

File: gcloud/storage/blob.py [hunk 1]
```diff
@@ -346,6 +347,16 @@ def download_as_string(self, client=None):
         self.download_to_file(string_buffer, client=client)
         return string_buffer.getvalue()
 
+    @staticmethod
+    def _check_response_error(request, http_response):
+        """Helper for :meth:`upload_from_file`."""
+        info = http_response.info
+        status = int(info['status'])
+        if not 200 <= status < 300:
+            faux_response = _FauxResponse(status)
```
Comment (dhermes) [inline gcloud/storage/blob.py]: @tseaver you can just use

```python
response = httplib2.Response({'status': status})
```

Comment (tseaver) [inline gcloud/storage/blob.py]: d8729b7

**Comment by dhermes** [comment]
LGTM

**Comment by dhermes** [comment]
@tseaver This is #1717. Which issue is this supposed to close?

**Comment by dhermes** [comment]
Ahhh I see in the branch name #1712.

**Pull request merged**